src/main/java/io/elastic/jdbc/actions/NewSelectAction.java

```diff
@@ -0,0 +1,132 @@
+package io.elastic.jdbc.actions;
+
+import io.elastic.api.EventEmitter;
+import io.elastic.api.ExecutionParameters;
+import io.elastic.api.Function;
+import io.elastic.api.Message;
+import io.elastic.jdbc.query_builders.Query;
+import io.elastic.jdbc.utils.QueryFactory;
+import io.elastic.jdbc.utils.Utils;
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Map;
+import javax.json.Json;
+import javax.json.JsonArrayBuilder;
+import javax.json.JsonObject;
+import javax.json.JsonObjectBuilder;
+import javax.json.JsonString;
+import javax.json.JsonValue;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+public class NewSelectAction implements Function {
+  private static final Logger LOGGER = LoggerFactory.getLogger(NewSelectAction.class);
+  private static final String SQL_QUERY_VALUE = "sqlQuery";
+  private static final String PROPERTY_ALLOW_ZERO_RESULT = "allowZeroResults";
+  private static final String EMIT_BEHAVIOUR = "emitBehaviour";
+
+  @Override
+  public void execute(ExecutionParameters parameters) {
+    JsonObject body = parameters.getMessage().getBody();
+    final JsonObject configuration = parameters.getConfiguration();
+    EventEmitter eventEmitter = parameters.getEventEmitter();
+    checkConfig(configuration);
+    String dbEngine = configuration.getString("dbEngine");
+    String sqlQuery = configuration.getString("sqlQuery");
+    String emitBehaviour = "emitIndividually";
+
+    try {
+      emitBehaviour = configuration.getString(EMIT_BEHAVIOUR);
+    } catch (NullPointerException e) {
+      LOGGER.info("No Emit behavior is specified, the default value Emit Individually will be used");
+    }
+
+    Utils.columnTypes = Utils.getVariableTypes(sqlQuery);
```

Comment:
That smalls bad. Violation of open-closed SOLID principle

Reply:
just copy past from existing actions